airflow/models/dag_edge.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from sqlalchemy import Column, String, Index, Integer
+
+from airflow.utils.db import create_session
+
+from airflow.models.base import Base
+
+
+class DagEdge(Base):
+    """
+    This will store DAG edges. This will be the upstream and downstream dependencies of task inside a DAG.
+    """
+
+    __tablename__ = "dag_edge"
+
+    dag_id = Column(String(250), primary_key=True)
+    graph_id = Column(Integer, primary_key=True)
+    task_from = Column(String(250), primary_key=True)
+    task_to = Column(String(250), primary_key=True)
+
+    __table_args__ = (Index('idx_dag_edge', dag_id, graph_id, unique=False),)
```

Comment:
given both dag_id and graph_id are primary_key, why do we need another compound index here?

Reply:
There was some discussions about this above, not sure what to do with this now. I think that some database implementations do benefit from this index. Maybe someone has some comments about this?

Reply:
The PK is (dag_id,graph_id,task_from,task_to) - but often we will want to get all the edges for a graph - i.e. run `SELECT * form dag_edge WHERE dag_id = ? AND graph_id = ?` so having an index on just those to columns will help every database.